Sources/Storage/Disk/Config/DiskConfigurable.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  DiskConfigurable.swift
+//  StatsdClient
+//
+//  Created by Nghia Tran on 10/8/17.
+//  Copyright © 2017 StatsdClient. All rights reserved.
+//
+
+import Foundation
+
+protocol DiskConfigurable {
+
+    // Name
+    // It's also name of folder cache
+    var name: String { get }
+
+    // Max size on bytes
+    var maxSize: UInt { get }
+
+    // Cache folder type
+    var directoryType: DirectoryType { get }
+
+    // Protection mode in iOS/tvOS
+    var protectionType: FileProtectionType? { get }
+}
+
+extension DiskConfigurable {
+
+    // Default folder Type
+    var directoryType: DirectoryType {
+        return .cache
+    }
+
+    // Default cache folder
+    var directory: URL? {
+        return try? FileManager.default.url(for: directoryType.toSearchPath,
+                                            in: .userDomainMask,
+                                            appropriateFor: nil,
+                                            create: true)
+    }
+
+    // Path folder by directory + name
+    var pathFolder: String? {
```

Comment:
Add this to the protocol requirement, a `DiskConfigurable` is useless without a directory to save files to. and rename it to `directoryPath`

Reply:
I don't think `var directory: URL?` variable should be required again because we already had `directoryType`, which is required by `DiskConfigurable` protocol.
Furthermore `directoryType ` explicitly indicates where it should be saved.
If we require both, it's redundant. 
We could create `directory` from `directoryType ` enum easily. It's the eason why I put `directory ` as a helper variable.

How do you think?

Reply:
rename the variable to `directoryPath`